keras/src/callbacks/orbax_checkpoint_test.py

```diff
@@ -577,3 +587,801 @@ def compare_nested_dicts(orig_dict, loaded_dict):
             original_state_tree["metrics_variables"],
             loaded_state_tree["metrics_variables"],
         )
+
+    @pytest.mark.requires_trainable_backend
+    def _flatten_nested_dict(self, nested_dict):
+        """Flatten a nested dictionary into a flat dictionary with path keys."""
+        flat_dict = {}
+
+        def _flatten(current_dict, prefix=""):
+            for key, value in current_dict.items():
+                if isinstance(value, dict):
+                    _flatten(value, f"{prefix}{key}/")
+                else:
+                    flat_dict[f"{prefix}{key}"] = value
+
+        _flatten(nested_dict)
+        return flat_dict
```

Comment:
This already exists, it's `tree.flatten_with_path` (`from keras.src import tree`), you'll get a list of tuple with path as a tuple and value.

So `_flatten_nested_dict` is equivalent to:
```python
{".".join(p):v for p, v in tree.flatten_with_path(nested_dict)}
```